src/view/components/Tab/RestaurantListTab.ts

```diff
@@ -0,0 +1,110 @@
+      tabMenu: this.#generateTabMenu(),
+      contentsArea: document.getElementById("restaurants-container")!,
+    });
+    this.#styledCurrentTab();
+  }
+
+  #generateTabMenu() {
+    return {
+      "모든 음식점": this.#sectionsForAllList(),
+      "자주 가는 음식점": this.#sectionsForFavoriteList(),
+    };
+  }
+
+  #sectionsForAllList() {
+    const filterContainer = createElementByTag({
+      tag: "section",
+      classes: ["restaurant-filter-container"],
+      attribute: { id: "filter-container" },
+    });
+
+    const restaurantListContainer = createElementByTag({
+      tag: "section",
+      classes: ["restaurant-list-container"],
+      attribute: { id: "restaurant-all-list-container" },
+    });
+
+    restaurantListContainer.addEventListener("click", onClickHandle);
+
+    return [filterContainer, restaurantListContainer];
+  }
+
+  #sectionsForFavoriteList() {
+    const restaurantListContainer = createElementByTag({
+      tag: "section",
+      classes: ["restaurant-list-container"],
+      attribute: { id: "restaurant-favorite-list-container" },
+    });
+
+    restaurantListContainer.addEventListener("click", onClickHandle);
+    return [restaurantListContainer];
+  }
+
+  #styledCurrentTab() {
+    const currentTab = this.tabElement.getAttribute("value")!;
+    const tabButton = this.tabElement.firstChild as HTMLButtonElement;
+    this.#handleCurrentTabClassName(currentTab, tabButton);
+
+    this.tabElement.addEventListener("click", (e) => {
+      if (
+        !(
+          e.currentTarget instanceof HTMLElement &&
+          e.target instanceof HTMLElement
+        )
+      ) {
+        throw new Error(
+          "[ERROR_IN_styledCurrentTab] e.currentTarget Or e.target is not HTMLElement"
+        );
+      }
+
+      const currentTab = e.currentTarget.getAttribute("value")!;
+      e.currentTarget.childNodes.forEach((tabButton) => {
+        this.#handleCurrentTabClassName(
+          currentTab,
+          tabButton as HTMLButtonElement
+        );
+      });
+    });
+  }
+
+  #handleCurrentTabClassName(currentTab: string, tabButton: HTMLButtonElement) {
+    if (currentTab === tabButton.id.slice(TAB_ID_PREFIX.length)) {
+      tabButton.classList.add("currentTab");
+      return;
+    }
+    tabButton.classList.remove("currentTab");
+  }
+}
+
+const onClickHandle = (e: MouseEvent) => {
+  if (
+    e.target instanceof HTMLElement &&
+    !(e.target.parentElement instanceof HTMLButtonElement)
+  ) {
+    const restaurantName = e.target.closest("li")?.getAttribute("name");
+    if (!restaurantName) {
+      throw new Error(
+        "[ERROR_IN_RestaurantListTab_onClickHandle()] Can't find valid restaurantName"
+      );
+    }
+    const detail = getRestaurant(restaurantName);
+    const detailModal = new RestaurantDetailModal(detail!);
+    document
+      .getElementById("restaurant-detail-modal-container")
+      ?.replaceChildren(detailModal.element);
+    detailModal.toggle();
+  }
+};
```

Comment:
if 중첩문이 깊어질수록 코드를 읽어갈 때 생각해야 하는 상위 조건들이 많아지고, 그에 따라 코드의 가독성이 떨어지는데요. 적절하게 early return을 활용해보셔도 좋을거 같아요
```suggestion
const onClickHandle = (e: MouseEvent) => {
  if (
    !e.target instanceof HTMLElement ||
    e.target.parentElement instanceof HTMLButtonElement
  ) return;
  
  
  const restaurantName = e.target.closest("li")?.getAttribute("name");
  if (!restaurantName) {
      throw new Error(
        "[ERROR_IN_RestaurantListTab_onClickHandle()] Can't find valid restaurantName"
      );
   }
   
    const detail = getRestaurant(restaurantName);
    const detailModal = new RestaurantDetailModal(detail!);
    document
      .getElementById("restaurant-detail-modal-container")
      ?.replaceChildren(detailModal.element);
    detailModal.toggle();
  }
};
```

Reply:
한번에 많은 정보들을 기억하며 코드를 읽다보면 쉽게 인지 부하가 오고 코드의 내용을 잘못 이해할 확률이 올라가는데요. 또 다른 예시로 let을 사용하는 경우 let으로 선언된 변수가 어떻게 변화되는지 코드블럭 내내 신경쓰며 읽어야 하고 이는 코드를 읽기 어렵게 만듭니다. 코드를 작성하실 때 이런 부분을 같이 고민해보시면 좋겠습니다 ㅎㅎ 👍

Reply:
가독성을 높이고 유지보수성을 높이기 위해서 함수를 짧게 유지해야하는데, 시간에 쫓긴다는 이유로 이런 부분에 대해 신경을 많이 쓰지 못했던 것 같습니다. 여전히 부족하긴 하지만, 유조가 리뷰해주신대로 early return을 활용해보았습니다!

👇 아래 커밋에 반영되었습니다!
[2ea2124](https://github.com/woowacourse/javascript-lunch/pull/161/commits/2ea2124181f26a042d0482f3406e7d27e80a3652)